minichain/rpc.py

```
@@ -0,0 +1,87 @@
+        logger.info("🚀 JSON-RPC Server running on http://%s:%d", host, port)
+
+    async def handle_rpc(self, request):
+        try:
+            req_data = await request.json()
+        except json.JSONDecodeError:
+            return web.json_response({"jsonrpc": "2.0", "error": {"code": -32700, "message": "Parse error"}, "id": None})
+
+        if isinstance(req_data, list):
+            responses = []
+            for req in req_data:
+                responses.append(await self._process_single(req))
```

Ruff (0.15.17)

[warning] Use a list comprehension to create a transformed list

(PERF401)